osgar/lib/unittest.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+
+import numpy as np
+
+# if this variable is set, stack trace on test failure from this module is not in the output
+__unittest = True
+
+class TestCase(unittest.TestCase):
+
+    def __init__(self, methodName='runTest'):
+        super().__init__(methodName)
+        self.addTypeEqualityFunc(np.ndarray, 'assertNumpyEqual')
+
+    def assertNumpyEqual(self, n1, n2, msg=None):
+        self.assertIsInstance(n1, np.ndarray, 'First argument is not a numpy.ndarray')
+        self.assertIsInstance(n2, np.ndarray, 'Second argument is not a numpy.ndarray')
+        np.testing.assert_almost_equal(n1, n2, msg)
+
+    def assertPose3dEqual(self, p1, p2):
+        self.assertEqual(len(p1), 2, msg='First Pose3d needs xyz and quaternion')
+        self.assertEqual(len(p2), 2, msg='Second Pose3d needs xyz and quaternion')
+        self.assertListAlmostEqual(p1[0], p2[0])
+        self.assertListAlmostEqual(p1[1], p2[1])
+
+    def assertListAlmostEqual(self, l1, l2, places=None, delta=None):
```

Comment:
this could be compatible to Python3 almost equal:
`assertAlmostEqual(first, second, places=7, msg=None, delta=None)`
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotAlmostEqual
? but OK

Reply:
I had the `msg` argument there but I didn't know what to do with it, since I give custom generate `msg` to the asserts called inside.